### PR TITLE
doc: for usePublic true on the Cloud

### PR DESCRIPTION
**Pull request by sancar**

### CODE REVIEW

**Comment by sancar** [comment]
run-lab-run

File: hazelcast/src/main/java/com/hazelcast/client/impl/clientside/ClusterDiscoveryServiceBuilder.java [hunk 1]
```diff
@@ -139,6 +139,9 @@ private AddressProvider createAddressProvider(ClientConfig clientConfig, Discove
             String urlEndpoint = HazelcastCloudDiscovery.createUrlEndpoint(cloudUrlBase, discoveryToken);
             int connectionTimeoutMillis = getConnectionTimeoutMillis(networkConfig);
             HazelcastCloudDiscovery cloudDiscovery = new HazelcastCloudDiscovery(urlEndpoint, connectionTimeoutMillis);
+            //We use the usePublic parameter as true always because on the cloud context hazelcast nodes are never in the
+            //same network even-tough they can be in the same group/zone etc. For private addresses to be used, the
+            //members and clients should be all coordinated by the users so that private addresses can be used.
```
Comment (ihsandemir) [inline hazelcast/src/main/java/com/hazelcast/client/impl/clientside/ClusterDiscoveryServiceBuilder.java:144]: what does this mean? 
```
//members and clients should be all coordinated by the users
```

Comment (sancar) [reply hazelcast/src/main/java/com/hazelcast/client/impl/clientside/ClusterDiscoveryServiceBuilder.java:144]: I tried to meant all the machines that contains clients and members needs to be opened by the same organization with same credentials to make private addresses work between them.

If this is is more clear, I will update the description accordingly.

Comment (ihsandemir) [reply hazelcast/src/main/java/com/hazelcast/client/impl/clientside/ClusterDiscoveryServiceBuilder.java:144]: yes, please clarify it but even the explanation you put sounds very complex, can you simplify it? Why do you even need that sentence? Is the first sentence not enough?

**Review by utku-caglayan — APPROVED** [review]

File: hazelcast/src/main/java/com/hazelcast/client/impl/clientside/ClusterDiscoveryServiceBuilder.java (hunk 1 above)
Comment (sancar) [reply hazelcast/src/main/java/com/hazelcast/client/impl/clientside/ClusterDiscoveryServiceBuilder.java:144]: Ok, removing the second sentence.

**Pull request merged**